File: 0412.FizzBuzz.go
func fizzBuzz(n int) []string {
    // Not using "%" operation
    ans := make([]string, n)
    fizz, buzz := 3, 5
    for i := 1; i <= n; i++ {
        if i == fizz && i == buzz {
            ans[i-1] = "FizzBuzz"
            fizz += 3
            buzz += 5
        } else if i == fizz {
            ans[i-1] = "Fizz"
            fizz += 3
        } else if i == buzz {
            ans[i-1] = "Buzz"
            buzz += 5
        } else {
            ans[i-1] = strconv.Itoa(i)
        }
    }
    return ans
}

File: 0412.FizzBuzz.py
class Solution(object):
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        return [str(i) if i%3 != 0 and i%5 != 0 else 'Fizz'*(i%3==0) + 'Buzz'*(i%5==0) for i in xrange(1, n+1)]
        ans = []
        for i in xrange(1, n+1):
            tmp = ''
            if i%3 == 0:
                tmp = 'Fizz'
            if i%5 == 0:
                tmp += 'Buzz'
            if not tmp:
                tmp = str(i)
            ans.append(tmp)
        return ans
        